src/pages/overlay/components/overlay/Overlay.tsx

```diff
@@ -164,6 +164,35 @@ export default function Overlay() {
     return () => document.body.removeEventListener("click", bodyClick, true);
   }, [bodyClick]);
 
+  // Handle body double clicks, ignoring them inside of overlay elements
+  const bodyDblClick = useCallback((e: MouseEvent) => {
+    // Get all the elements under the mouse
```

Comment:
Can we abstract this logic for finding a visible element under the mouse, given both this handler and the regular click handler are doing almost identical things? Maybe a `visibleUnderCursor` method that turns the first visible element? And then both the handlers could just call that in an if, instead of duplicating the for loop etc.